core/src/LuminanceSource.cpp

```diff
@@ -41,14 +41,16 @@ class InvertedLuminanceSource : public LuminanceSource
 	const uint8_t* getRow(int y, ByteArray& outBytes, bool) const override
 	{
 		_src->getRow(y, outBytes, true);
-		std::transform(outBytes.begin(), outBytes.end(), outBytes.begin(), [](uint8_t b) { return 255 - b; });
+		for (auto& b : outBytes)
+			b = 255 - b;
 		return outBytes.data();
```

Comment:
You could argue that the for loop is more readable but we generally went with the "Sean Parent" approach of avoiding raw loops if possible to increase reasonability/readability. Based on your comment, you seem to think the for loop is faster? To believe that I would have to be convinced with some assembler output or measurements. :)

Reply:
Well, it's a range-based loop (a modern C++ citizen), which just expresses the intention to access and possibly modify (the `auto&` stuff) each element.
I went with the rule of the thumb to prefer references over the copies (i.e. the argument of the lambda), even if compilers would certainly optimize this on themselves.
Anyway, all my proposal are... well, just proposals! :)

Reply:
My point was: 1) performance is the exact same and 2) your option is actually more readable but less consistent with the rest. I'm trying to get comfortable with std algorithms, so would rather leave it as is. @huycn: we need a third opinion ;).

Reply:
Zut... I was trying to avoid the topic ;) 

To me, both are valid, of question of styles. `std::transform()` is a little bit harder to read at first, but it's not an issue once you get used to it. Personally, I still find the `for` loop is more readable, especially when once switches often between different languages at work (foreach exists in almost all languages I known). At the same time, the `std::transform` has an advantage of functional programming and more consistent with the rest.

So it will not be clean cut here and in such case, I normally go with "don't fix if it ain't broke" principle.

Reply:
OK, no problem, also where I come from any modification requires at least 50%+1 of the consensus.

Questions:
- am I allowed to add pragma warning disable directives to library code?
- am I allowed to wrap them in MSVC specific enabler and hope someone else would take care of introduce other compiler enablers (if ever needed)?

Reply:
I believe the correct location to add diagnostic modifying flags is the CMakeLists.txt. See there for other examples. And if you wander why there are currently duplicates of those kind of settings, see also #86. So my answer to your 3 questions would be no, not applicable and no.

Reply:
Done in 6dc5997d37b4ae7d691a7326be4eb1ad2f97510a, anyway I'll wait for a feedback.